pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go

```diff
@@ -0,0 +1,645 @@
+		}
+		networkPolicy.Spec = networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					v1beta1constants.GardenRole: "controlplane",
+					v1beta1constants.LabelApp:   "vpn-seed-server",
+				},
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{
+				{
+					From: []networkingv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									v1beta1constants.GardenRole: "controlplane",
+									v1beta1constants.LabelApp:   "kubernetes",
+									v1beta1constants.LabelRole:  "apiserver",
+								},
+							},
+						},
+					},
+				},
+				{
+					From: []networkingv1.NetworkPolicyPeer{
+						{
+							// we don't want to modify existing labels on the istio namespace
+							NamespaceSelector: &metav1.LabelSelector{},
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									v1beta1constants.LabelApp: "istio-ingressgateway",
+								},
+							},
+						},
+					},
+				},
+			},
+			Egress: []networkingv1.NetworkPolicyEgressRule{
+				{
+					To: []networkingv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									v1beta1constants.GardenRole: "controlplane",
+									v1beta1constants.LabelApp:   "kubernetes",
+									v1beta1constants.LabelRole:  "apiserver",
+								},
+							},
+						},
+					},
+				},
+			},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeIngress,
+				networkingv1.PolicyTypeEgress,
+			},
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, deployment, func() error {
+		maxSurge := intstr.FromInt(100)
+		maxUnavailable := intstr.FromInt(0)
+		deployment.Labels = map[string]string{
+			v1beta1constants.GardenRole:                           "controlplane",
+			v1beta1constants.LabelApp:                             DeploymentName,
+			v1beta1constants.LabelNetworkPolicyFromShootAPIServer: v1beta1constants.LabelNetworkPolicyAllowed,
+		}
+		deployment.Spec = appsv1.DeploymentSpec{
+			MinReadySeconds:      30,
+			Replicas:             pointer.Int32Ptr(v.replicas),
+			RevisionHistoryLimit: pointer.Int32Ptr(1),
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{
+				v1beta1constants.LabelApp: DeploymentName,
+			}},
+			Strategy: appsv1.DeploymentStrategy{
+				RollingUpdate: &appsv1.RollingUpdateDeployment{
+					MaxUnavailable: &maxUnavailable,
+					MaxSurge:       &maxSurge,
+				},
+				Type: appsv1.RollingUpdateDeploymentStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						v1beta1constants.GardenRole:                          "controlplane",
+						v1beta1constants.DeprecatedGardenRole:                "controlplane",
```

Comment:
See above.
BTW, any reason why you set the `DeprecatedGardenRole` here? If not, remove it please.

Reply:
No particular reason. We just used a template, which had both.
The deprecated part was removed in the last commit.